server/src/main/java/io/druid/server/coordination/DruidServerMetadata.java

```diff
@@ -40,7 +43,10 @@ public DruidServerMetadata(
       @JsonProperty("maxSize") long maxSize,
       @JsonProperty("type") String type,
       @JsonProperty("tier") String tier,
-      @JsonProperty("priority") int priority
+      @JsonProperty("priority") int priority,
+      @JsonProperty("service") String service,
+      @JsonProperty("hostText") String hostText,
```

Comment:
I think the confusion comes form the fact that we currently use CuratorServiceAnnouncer for both internal and external announcement, which writes DruidNode into Zookeeper instead of DruidServerMetadata, though it should've written DruidServerMetadata since the later one is intended to contain all the necessary information. 

The reason I have to add ```hostText```, ```port``` and ```service``` is because those information are  in DruidNode but not in DruidServerMetadata (though it should've been). I think in the future we could deprecate ```host``` since it's basically the concatenation of ```hostText``` + ```port```.